### ISSUE TITLE
Add a method to apply transparency on rendered result instead of individual components

### ISSUE THREAD

**Issue by Xascoria** [godotengine/godot-proposals#1670]
**Describe the project you are working on:**
n/a

**Describe the problem or limitation you are having in your project:**

It is quite a common problem that when rendering with transparency, shadows, and outlines, the wanted result is too combine all the visible component in an non-opaque state and then give them transparency and shadows instead of giving them individually, which would create overlapping areas and exposes unwanted details in the final result.

Here's an example from a Unity shader to illustrate my point:
https://ocias.com/blog/wp-content/uploads/2016/06/UnityIssues-1.gif

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**

Allow graphical components to be grouped and rendered together first, then be applied effects such as outline or transparency.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**

Maybe allow a node to have a setting where all its children are rendered as one component and then whatever effects that's applied to the parent is applied to all of them at once, or maybe there's a better way but this is what I can think of now.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**

There is workaround, which involves relying on layering viewports and then apply transparency/outline on the rendered result, but that's not something that can be done in a few lines of script and super over-killed when all you need is to group a couple objects together.

**Is there a reason why this should be core and not an add-on in the asset library?:**

It is a really common problem. See https://cdn.discordapp.com/attachments/212250894228652034/766234248759083058/unknown.png

**Comment by Calinou** [comment]
It's a common issue, but I don't think we can efficiently expose a simple way to do this. All we can do is better document how to use a viewport to achieve this.

**Comment by Xascoria** [comment]
Not my screenshot, blame Zylann for that

As for determine what should be "grouped" together in alpha blending, I think a parent node grouping all its children and itself together is a pretty effective way for that

**Comment by Xrayez** [comment]
> better document how to use a viewport to achieve this.

This also needs a plugin, it's pretty cumbersome to do this and the process mostly involves trial-and-error approach (aligning the sprite size to fit the viewport etc.). Usability issues like godotengine/godot#19239 are inherent to viewport usage.

The way I see it, you'd select a bunch of nodes (a branch?) and click on "Convert/merge into a Viewport". 😃

**Comment by Xrayez** [comment]
Would godotengine/godot#43167 help this in Godot 4.0?

**Comment by clayjohn** [comment]
@Xrayez Yes! But only in 2D. It appears the OP wants this for 2D and 3D.
